x-pack/plugins/alerting/server/authorization/alerting_authorization.ts

```diff
@@ -351,6 +366,28 @@ export class AlertingAuthorization {
                   ruleTypeAuth.producer === feature,
                 ]
               );
+              if (!isEmpty(ruleTypeAuth.validLegacyConsumers)) {
```

Comment:
Can we add some code comments to explain what this is doing?

Reply:
I will add a comment!!!